src/compiler/checker.ts

```diff
@@ -24223,6 +24223,12 @@ namespace ts {
                     }
                 }
 
+                const firstFunctionLikeContainerOrClassLikeContainer = findAncestor(node.parent, node => isClassLike(node) || isFunctionLike(node))!;

```

Comment:
This is not correct. Consider this code:

```js
function a() {
    return (new class { a() { return arguments } }).a()
}
```

It's valid in JavaScript. Please make sure you check the arguments is in the banned position of a class member